src/crucible/aws/ecs/task_definition.clj

```diff
@@ -0,0 +1,40 @@
+(ns crucible.aws.ecs.task-definition
+  "Resources in AWS::ECS::TaskDefinition"
+  (:require [clojure.spec.alpha :as s]
+            [crucible.resources :refer [spec-or-ref defresource]]
+            [crucible.aws.ecs.container-definition :as cd]))
+
+(s/def ::container-definitions (s/coll-of ::cd/container-definition :kind vector?))
+
+(s/def ::cpu (spec-or-ref string?))
```

Comment:
Could be more specific:

(s/def ::cpu #{256 512 1024 2048 4096})

Reply:
For a Fargate launch type you are correct and are limited to the above values as strings however for a normal ec2 user the value is optional and can have any value. I'm not sure how to make the spec conditional on the launch type though.

Reply:
Ah right, good catch. I'm on FARGATE obviously ;-)
There are more specs that can be made conditionally based on other values, we'll have to learn spec a bit better. You are obviously already doing better than I am.